core/controller.go

```diff
@@ -268,6 +303,51 @@ func (c *Controller) StopStreamHandler(w http.ResponseWriter, r *http.Request, p
 	w.WriteHeader(http.StatusOK)
 }
 
+func (c *Controller) StartPreloadStream(Name string, Uri string) {
+	var dto StreamDTO
```

Comment:
We don't need this dto here. We can just use the URI parameter
```go
    _, knownStream := c.index[URI]
	if knownStream {
		return
	}

	stream, id := streamer.NewStream(
		URI,
		c.spec.StoreDir,
		c.spec.KeepFiles,
		c.spec.Audio,
		streamer.ProcessLoggingOpts{
			Enabled:    c.spec.ProcessLogging.Enabled,
			Compress:   c.spec.ProcessLogging.Compress,
			Directory:  c.spec.ProcessLogging.Directory,
			MaxAge:     c.spec.ProcessLogging.MaxAge,
			MaxBackups: c.spec.ProcessLogging.MaxBackups,
			MaxSize:    c.spec.ProcessLogging.MaxSize,
		},
		25*time.Second,
	)
```

Reply:
changed